core/trino-parser/src/main/antlr4/io/trino/sql/parser/SqlBase.g4

```diff
@@ -418,6 +418,54 @@ relationPrimary
     | LATERAL '(' query ')'                                           #lateral
     | TABLE '(' tableFunctionCall ')'                                 #tableFunctionInvocation
     | '(' relation ')'                                                #parenthesizedRelation
+    | JSON_TABLE '('
+        jsonPathInvocation
+        COLUMNS '(' jsonTableColumn (',' jsonTableColumn)* ')'
+        (PLAN '(' jsonTableSpecificPlan ')'
+        | PLAN DEFAULT '(' jsonTablePlanDefault ')')?
+        ((ERROR | EMPTY) ON ERROR)?
+      ')'                                                             #jsonTable
+    ;
+
+jsonTableColumn
+    : identifier FOR ORDINALITY                                     #ordinalityColumn
+    | identifier type
+        (PATH string)?
+        (emptyBehavior=jsonValueBehavior ON EMPTY)?
+        (errorBehavior=jsonValueBehavior ON ERROR)?                 #valueColumn
+    | identifier type FORMAT jsonRepresentation
+        (PATH string)?
+        (jsonQueryWrapperBehavior WRAPPER)?
+        ((KEEP | OMIT) QUOTES (ON SCALAR TEXT_STRING)?)?
+        (emptyBehavior=jsonQueryBehavior ON EMPTY)?
+        (errorBehavior=jsonQueryBehavior ON ERROR)?                 #queryColumn
+    | NESTED PATH? string (AS identifier)?
+        COLUMNS '(' jsonTableColumn (',' jsonTableColumn)* ')'      #nestedColumns
+    ;
+
+jsonTableSpecificPlan
+    : leafPlan
+    | compositePlan
+    ;
+
+leafPlan
+    : identifier
+    ;
+
+compositePlan
```

Comment:
This is unnecessary. You should be able to condense all these productions into something like:

```
jsonTablePlan
    : jsonTablePathName
    | jsonTablePathName (OUTER | INNER) planPrimary
    | planPrimary UNION planPrimary (UNION planPrimary)*
    | planPrimary CROSS planPrimary (CROSS planPrimary)*
    ;

jsonTablePathName
    : identifier
    ;

planPrimary
    : jsonTablePathName
    | '(' jsonTablePlan ')'
    ;
```